formats/json/commonMain/src/kotlinx/serialization/json/internal/TreeJsonEncoder.kt

```diff
@@ -155,7 +158,14 @@ private sealed class AbstractJsonTreeEncoder(
 
         val discriminator = polymorphicDiscriminator
         if (discriminator != null) {
-            encoder.putElement(discriminator, JsonPrimitive(polymorphicSerialName ?: descriptor.serialName))
+            if (encoder is JsonTreeMapEncoder) {
+                // first parameter is ignored in JsonTreeMapEncoder
```

Comment:
I don't get what 'first parameter' you are referring to

Reply:
First parameter in implementation of `putElement` for `JsonTreeMapEncoder`
```
    override fun putElement(key: String, element: JsonElement) {
        if (isKey) { // writing key
            tag = when (element) {
                is JsonPrimitive -> element.content
                is JsonObject -> throw InvalidKeyKindException(JsonObjectSerializer.descriptor)
                is JsonArray -> throw InvalidKeyKindException(JsonArraySerializer.descriptor)
            }
            isKey = false
        } else {
            content[tag] = element
            isKey = true
        }
    }
```
`key` - is unused